src/main/java/org/folio/domain/ActionBlocks.java

```diff
@@ -22,68 +23,83 @@
   private final boolean blockRequests;
 
-  public static ActionBlocks byLimit(UserSummary userSummary, PatronBlockLimit limit) {
-    if (userSummary == null || limit == null || userSummary.getOpenLoans() == null) {
-      log.error("Failed to determine blocks: one of the parameters is null");
+  public static ActionBlocks byLimit(UserSummary userSummary, PatronBlockLimit patronBlockLimit) {
+    log.debug("byLimit:: parameters userSummary: {}, patronBlockLimit: {}",
+      () -> logAsJson(userSummary), () -> patronBlockLimit);
+    if (userSummary == null || patronBlockLimit == null || userSummary.getOpenLoans() == null) {
+      log.warn("byLimit:: Failed to determine blocks because one of the parameters is null; " +
+        "parameters userSummary: {}, patronBlockLimit: {}", () -> logAsJson(userSummary),
+        () -> patronBlockLimit);
       return empty();
     }
-
-    return byLimit(userSummary, limit, userSummary.getOpenLoans().stream()
+    ActionBlocks actionBlocks = byLimit(userSummary, patronBlockLimit, userSummary.getOpenLoans().stream()
       .filter(openLoan -> openLoan.getLoanId() != null)
       .collect(Collectors.toMap(OpenLoan::getLoanId, r -> 0)));
+    log.info("byLimit:: result: {}", () -> logAsJson(actionBlocks));
+    return actionBlocks;
   }
 
-  public static ActionBlocks byLimit(UserSummary userSummary, PatronBlockLimit limit,
+  public static ActionBlocks byLimit(UserSummary userSummary, PatronBlockLimit patronBlockLimit,
     Map<String, Integer> overdueMinutes) {
-
-    if (userSummary == null || limit == null || overdueMinutes == null || limit.getValue() == null
-      || limit.getConditionId() == null) {
-
-      log.error("Failed to determine blocks: one of the parameters is null");
+    log.debug("byLimit:: parameters userSummary: {}, patronBlockLimit: {}, " +
+      "overdueMinutes: {}", () -> logAsJson(userSummary), () -> patronBlockLimit,
+      () -> overdueMinutes);
+    if (userSummary == null || patronBlockLimit == null || overdueMinutes == null ||
+      patronBlockLimit.getValue() == null || patronBlockLimit.getConditionId() == null) {
+      log.warn("byLimit:: Failed to determine blocks because one of the parameters is null; " +
+        "parameters userSummary: {}, patronBlockLimit: {}, patronBlockLimit.value, " +
+        "patronBlockLimit.conditionId), overdueMinutes: {}", () -> logAsJson(userSummary),
+        () -> patronBlockLimit, () -> overdueMinutes);
       return empty();
     }
 
-    Condition condition = Condition.getById(limit.getConditionId());
+    Condition condition = Condition.getById(patronBlockLimit.getConditionId());
 
     boolean blockBorrowing = false;
     boolean blockRenewals = false;
     boolean blockRequests = false;
 
-    double limitValue = limit.getValue();
+    double limitValue = patronBlockLimit.getValue();
 
     if (condition == MAX_NUMBER_OF_ITEMS_CHARGED_OUT) {
+      log.info(LOG_TEMPLATE_BY_LIMIT_CONDITION, condition);
```

Comment:
How about logging the condition once above all if's?

Reply:
The goal of logging if/else/switch is to understand which branch of code is being evaluated